google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -0,0 +1,157 @@
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.monitoring.v3.CreateTimeSeriesRequest;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.protobuf.Empty;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.threeten.bp.Duration;
+
+/** Bigtable Cloud Monitoring OpenTelemetry Exporter. */
+final class BigtableCloudMonitoringExporter implements MetricExporter {
+
+  private static final Logger logger =
+      Logger.getLogger(BigtableCloudMonitoringExporter.class.getName());
+  private final MetricServiceClient client;
+
+  private final String projectId;
+  private final String taskId;
+  private final MonitoredResource monitoredResource;
+  private final AtomicBoolean isShutdown = new AtomicBoolean(false);
+
+  private static final String RESOURCE_TYPE = "bigtable_client_raw";
+
+  private CompletableResultCode lastExportCode;
+
+  static BigtableCloudMonitoringExporter create(String projectId, Credentials credentials)
+      throws IOException {
+    MetricServiceSettings.Builder settingsBuilder = MetricServiceSettings.newBuilder();
+    settingsBuilder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+
+    org.threeten.bp.Duration timeout = Duration.ofMinutes(1);
+    // TODO: createServiceTimeSeries needs special handling if the request failed. Leaving
+    // it as not retried for now.
+    settingsBuilder.createServiceTimeSeriesSettings().setSimpleTimeoutNoRetries(timeout);
+    return new BigtableCloudMonitoringExporter(
+        projectId,
+        MetricServiceClient.create(settingsBuilder.build()),
+        MonitoredResource.newBuilder().setType(RESOURCE_TYPE).build(),
+        BigtableExporterUtils.getDefaultTaskValue());
+  }
+
+  @VisibleForTesting
+  BigtableCloudMonitoringExporter(
+      String projectId,
+      MetricServiceClient client,
+      MonitoredResource monitoredResource,
+      String taskId) {
+    this.client = client;
+    this.monitoredResource = monitoredResource;
+    this.taskId = taskId;
+    this.projectId = projectId;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> collection) {
+    if (isShutdown.get()) {
+      logger.log(Level.WARNING, "Exporter is shutting down");
+      return CompletableResultCode.ofFailure();
+    }
+    if (!collection.stream()
+        .flatMap(metricData -> metricData.getData().getPoints().stream())
+        .allMatch(pd -> BigtableExporterUtils.getProjectId(pd).equals(projectId))) {
```

Comment:
this might throw an NPE. Might be better to reverse this to projectId.equals(...)